### ISSUE TITLE
Auto generating interop C++/C# code.

### ISSUE THREAD

**Issue by pauldotknopf** [qmlnet/qmlnet#119]
Right now, it is cumbersome to flesh out the boilerplate needed for some simple getters/setters.

I'd like to have another console project where we can define the interop API with C# models, and then generate the C++/C# to go back and forth.

SWIG does someting similar, but I'd prefer a home-grown approach that uses ```QSharedPointer``` and special Qt meta-system support.

**Comment by pauldotknopf** [comment]
Note, this is for the internal interop only, not the classes defined in your implementation.

**Comment by vadi2** [comment]
Would this mean that any Qt class would be available for use, instead of only ones marked as `Q_INVOKABLE` right now?

**Comment by pauldotknopf** [comment]
No, unfortunately. The properties/methods that are dynamically invokable must be made visible to the meta object system. This is only done via Q_PROPERTY and Q_INVOKABLE (or slots).
